docs/design/aws_under_the_hood.md

```diff
@@ -0,0 +1,271 @@
+  enables the `dns-support` and `dns-hostnames` options.
+* Creates an internet gateway for the VPC.
+* Creates a route table for the VPC, with the internet gateway as the default
+  route
+* Creates a subnet (with a CIDR of 172.20.0.0/24) in the AZ `KUBE_AWS_ZONE`
+  (defaults to us-west-2a).  Currently kubernetes runs in a single AZ; there
+are two philosophies on how to achieve HA: cluster-per-AZ and
+cross-AZ-clusters.  cluster-per-AZ says you should have an independent cluster
+for each AZ, they are entirely separate.  cross-AZ-clusters allows a single
+cluster to span multiple AZs.  The debate is open here: cluster-per-AZ is more
+robust but cross-AZ-clusters are more convenient.  For now though, each AWS
+kuberentes cluster lives in one AZ.
+* Associates the subnet to the route table
+* Creates security groups for the master node (`kubernetes-master-<clusterid>`)
+  and the minion nodes (`kubernetes-minion-<clusterid>`)
+* Configures security groups so that masters & minions can intercommunicate,
+  and opens SSH to the world on master & minions, and opens port 443 to the
+world on the master (for the HTTPS API endpoint)
+* Creates an EBS volume for the master node of size `MASTER_DISK_SIZE` and type
+  `MASTER_DISK_TYPE`
+* Launches a master node with a fixed IP address (172.20.0.9), with the
+  security group, IAM credentials etc.  An instance script is used to pass
+vital configuration information to Salt.  The hope is that over time we can
+reduce the amount of configuration information that must be passed in this way.
+* Once the instance is up, it attaches the EBS volume & sets up a manual
+  routing rule for the internal network range (`MASTER_IP_RANGE`, defaults to
+10.246.0.0/24)
+* Creates an auto-scaling launch-configuration and group for the minions.  The
+  name for both is `<KUBE_AWS_INSTANCE_PREFIX>-minion-group`, defaults to
+`kubernetes-minion-group`.  The auto-scaling group has size min & max both set
+to `NUM_MINIONS`.  You can change the size of the auto-scaling group to add or
+remove minions (directly though the AWS API/Console).  The minion nodes
+self-configure: they come up, run Salt with the stored configuration; connect
+to the master and are assigned an internal CIDR; the master configures the
+route-table with the minion CIDR.  The script does health-check the minions,
+but this is a self-check, it is not required.
+
+If attempting this configuration manually, I highly recommend following along
+with the kube-up script, and being sure to tag everything with a
+`KubernetesCluster`=`<clusterid>` tag.  Also, passing the right configuration
+options to Salt when not using the script is tricky: the plan here is to
+simplify this by having Kubernetes take on more node configuration, and even
+potentially remove Salt altogether.
+
+
+## Manual infrastructure creation
+
+While this work is not yet complete, advanced users may choose to create (some)
```

Comment:
Edit (lines 232-233):
..., advanced users might choose to manually create AWS objects while still making use of the .....